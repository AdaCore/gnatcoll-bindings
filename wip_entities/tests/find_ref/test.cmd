!gnatmake -q main
PROJECT dummy.gpr

--  find all references to an entity
REFS proc1:main.adb:4:6
PARAMS proc1:main.adb:4:6

--  Same, but print list of subprograms that call entity
CALLERS proc1:main.adb:4:6

--  Same for another entity
CALLERS proc1:main.adb:5:6

