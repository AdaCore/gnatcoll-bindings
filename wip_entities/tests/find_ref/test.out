#!../../standalone_obj/test_parse
EXECUTING gnatmake -q main

--  find all references to an entity
REF: a.ads:2:14
REF: main.adb:4:6
REF: a.adb:4:14
REF: a.adb:7:8
REF: b.adb:5:9
Param (Proc1): A:a.ads:2:21 as IN_PARAMETER
Param (Proc1): C:a.ads:2:24 as IN_PARAMETER
Param (Proc1): F:a.ads:2:37 as IN_PARAMETER

--  Same, but print list of subprograms that call entity
CALLERS Proc1:a.ads:2:14
  REF: a.ads:2:14 => A:a.ads:1:9
  REF: a.adb:4:14 => A:a.ads:1:9
  REF: b.adb:5:9 => Proc1:b.ads:2:14
  REF: main.adb:4:6 => Main:main.adb:2:11

--  Same for another entity
CALLERS Proc1:b.ads:2:14
  REF: b.ads:2:14 => B:b.ads:1:9
  REF: a.adb:6:9 => Proc1:a.ads:2:14
  REF: a.adb:11:9 => Proc2:a.ads:3:14
  REF: b.adb:3:14 => B:b.ads:1:9
  REF: main.adb:5:6 => Main:main.adb:2:11
